src/firmware/mod.rs

```diff
@@ -128,6 +128,17 @@ impl From<std::io::Error> for Indeterminate<Error> {
     }
 }
 
+impl From<Indeterminate<Error>> for std::io::Error {
+    #[inline]
+    fn from(error: Indeterminate<Error>) -> std::io::Error {
+        if let Indeterminate::Known(Error::IoError(io_err)) = error {
+            return io_err;
+        }
+
+        std::io::ErrorKind::Other.into()
```

Comment:
Would you consider `impl std::error::Error for Error`? I think in that case you could take advantage of the `std::io::Error::new` constructor function as it takes a std::io::ErrorKind and any type that implements std::error::Error as a payload. Then you could just unpack the `Indeterminate::Known(blah)` and return `std::io::Error::new(ErrorKind::Other, blah)` Otherwise if it's Unknown you can use some other `std::io::ErrorKind`.